src/scenery_gui.cpp

```diff
@@ -43,6 +43,7 @@ class SceneryGui : public GuiWindow {
 	const SceneryType *selected_type;           ///< Currently selected item type.
 	uint8 orientation;                          ///< Current orientation.
 	std::unique_ptr<SceneryInstance> instance;  ///< Instance being placed.
+	StringID build_forbidden_reason;            ///< Reason why we may not place the instance at the given location, if any.
```

Comment:
I am guessing we're going to have more of these "compute max priority message"?

If you wrap a class around it, you can move some of the details into that class, eg
```
class BestReason {
public:
   ...
   StringId reason;
   void Reset() { reason = STR_NULL; }
   void UpdateMessage(StringID new_message) { CheckIfMoreImportant(&this->reason, new_message); }
```